src/tools/sss_override.c

```diff
@@ -81,14 +82,24 @@ static errno_t parse_cmdline(struct sss_cmdline *cmdline,
                               &orig_name, &domain);
     if (ret != EOK) {
         ERROR("Unable to parse name %s.\n", input_name);
-        return ret;
+        goto done;
     }
 
-    *_input_name = input_name;
+    *_input_name = strdup(input_name);
+    if (*_input_name == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
     *_orig_name = orig_name;
     *_domain = domain;
+    ret = EXIT_SUCCESS;
 
-    return EXIT_SUCCESS;
+done:
+    if (ret != EXIT_SUCCESS) {
```

Comment:
If `_input_name` is kept as a copy of `input_name`, then the latter should be freed unconditionally.